src/tcache.c

```diff
@@ -633,10 +635,24 @@ tcache_arena_reassociate(tsdn_t *tsdn, tcache_slow_t *tcache_slow,
 	tcache_arena_associate(tsdn, tcache_slow, tcache, arena);
 }
 
+static void
+tcache_max_and_nhbins_init(tcache_t *tcache) {
+	assert(tcache != NULL);
+	tcache->tcache_max = global_do_not_change_tcache_maxclass;
+	tcache->tcache_nhbins = global_do_not_change_nhbins;
+	assert(tcache->tcache_nhbins== sz_size2index(tcache->tcache_max) + 1);
```

Comment:
nit: missing space before `==`. Also assert `global_do_not_change_tcache_maxclass` and nhbins are not zero, to help detect init order issues.